Description: Build fixes
Author: Stephen Kitt <skitt@debian.org>

* Preserve build flags
* Echo commands
* Skip installing udev and systemd files

--- a/makefile
+++ b/makefile
@@ -2,10 +2,10 @@
 CXXFLAGS?=-Wall -O2
 LIB?=hidapi
 ifeq ($(LIB),libusb)
-	CPPFLAGS=-Dlibusb
+	CPPFLAGS+=-Dlibusb
 	LIBS=-lusb-1.0
 else
-	CPPFLAGS=-Dhidapi
+	CPPFLAGS+=-Dhidapi
 	LIBS=-lhidapi-hidraw
 endif
 SYSTEMDDIR?=/usr/lib/systemd
@@ -22,7 +22,9 @@
 
 CXXFLAGS+=-std=gnu++11 -DVERSION=\"$(MAJOR).$(MINOR).$(MICRO)\"
 APPSRCS=src/main.cpp src/helpers/*.cpp src/helpers/*.h
+APPOBJS := src/main.o $(patsubst %.cpp,%.o,$(wildcard src/helpers/*.cpp))
 LIBSRCS=src/classes/*.cpp src/classes/*.h
+LIBOBJS := $(patsubst %.cpp,%.o,$(wildcard src/classes/*.cpp))
 
 .PHONY: all bin debug clean setup install uninstall lib install-lib install-dev
 
@@ -30,90 +32,90 @@
 
 bin: bin/$(PROGN)
 
-bin/$(PROGN): $(APPSRCS) $(LIBSRCS)
-	@mkdir -p bin
-	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS) $^ -o $@ $(LIBS)
-	
+bin/$(PROGN): $(APPOBJS) $(LIBOBJS)
+	mkdir -p bin
+	$(CXX) $(LDFLAGS) $^ -o $@ $(LIBS)
+
 debug: CXXFLAGS += -g -Wextra -pedantic
 debug: bin/$(PROGN)
 
-lib/lib$(PROGN).so: $(LIBSRCS)
-	@mkdir -p lib
-	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS) -fPIC -shared -Wl,-soname,lib$(PROGN).so -o lib/lib$(PROGN).so.$(MAJOR).$(MINOR).$(MICRO) $^ $(LIBS)
-	@ln -sf lib$(PROGN).so.$(MAJOR).$(MINOR).$(MICRO) lib/lib$(PROGN).so
-
-bin-linked: lib/lib$(PROGN).so
-	@mkdir -p bin
-	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS) $(APPSRCS) -o bin/$(PROGN) $(LIBS) -L./lib -l$(PROGN)
+src/classes/%.o: src/classes/%.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -fPIC -c $^ -o $@
+
+lib/lib$(PROGN).so: $(LIBOBJS)
+	mkdir -p lib
+	$(CXX) $(LDFLAGS) -shared -Wl,-soname,lib$(PROGN).so.$(MAJOR) -o lib/lib$(PROGN).so.$(MAJOR).$(MINOR).$(MICRO) $^ $(LIBS)
+	ln -sf lib$(PROGN).so.$(MAJOR).$(MINOR).$(MICRO) lib/lib$(PROGN).so.$(MAJOR)
+	ln -sf lib$(PROGN).so.$(MAJOR) lib/lib$(PROGN).so
+
+bin-linked: lib/lib$(PROGN).so $(APPOBJS)
+	mkdir -p bin
+	$(CXX) $(LDFLAGS) $(APPOBJS) -o bin/$(PROGN) $(LIBS) -L./lib -l$(PROGN)
 
 lib: lib/lib$(PROGN).so
 
 clean:
-	@rm -rf bin
-	@rm -rf lib
+	rm -rf bin lib $(APPOBJS) $(LIBOBJS)
 
 setup:
-	@install -m 755 -d \
+	install -m 755 -d \
 		$(DESTDIR)/usr/bin \
 		$(DESTDIR)/etc/$(PROGN)/samples \
 		$(DESTDIR)/etc/udev/rules.d
-	@cp bin/$(PROGN) $(DESTDIR)/usr/bin
-	@test -s $(DESTDIR)/usr/bin/g213-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g213-led
-	@test -s $(DESTDIR)/usr/bin/g410-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g410-led
-	@test -s $(DESTDIR)/usr/bin/g413-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g413-led
-	@test -s $(DESTDIR)/usr/bin/g512-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g512-led
-	@test -s $(DESTDIR)/usr/bin/g513-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g513-led
-	@test -s $(DESTDIR)/usr/bin/g610-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g610-led
-	@test -s $(DESTDIR)/usr/bin/g910-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g910-led
-	@test -s $(DESTDIR)/usr/bin/gpro-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/gpro-led
-	@cp sample_profiles/* $(DESTDIR)/etc/$(PROGN)/samples
-	@cp udev/$(PROGN).rules $(DESTDIR)/etc/udev/rules.d
-	@test -s /usr/bin/systemd-run && \
-		install -m 755 -d $(DESTDIR)$(SYSTEMDDIR)/system && \
-		cp systemd/$(PROGN)-reboot.service $(DESTDIR)$(SYSTEMDDIR)/system
+	cp bin/$(PROGN) $(DESTDIR)/usr/bin
+	test -s $(DESTDIR)/usr/bin/g213-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g213-led
+	test -s $(DESTDIR)/usr/bin/g410-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g410-led
+	test -s $(DESTDIR)/usr/bin/g413-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g413-led
+	test -s $(DESTDIR)/usr/bin/g512-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g512-led
+	test -s $(DESTDIR)/usr/bin/g513-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g513-led
+	test -s $(DESTDIR)/usr/bin/g610-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g610-led
+	test -s $(DESTDIR)/usr/bin/g910-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/g910-led
+	test -s $(DESTDIR)/usr/bin/gpro-led || ln -s /usr/bin/$(PROGN) $(DESTDIR)/usr/bin/gpro-led
+	cp sample_profiles/* $(DESTDIR)/etc/$(PROGN)/samples
 
 install-lib: lib
-	@install -m 755 -d $(libdir)
-	@install -m 644 lib/lib$(PROGN).so.$(MAJOR).$(MINOR).$(MICRO) $(libdir)/
-	@ln -sf lib$(PROGN).so.$(MAJOR).$(MINOR).$(MICRO) $(libdir)/lib$(PROGN).so
+	install -m 755 -d $(libdir)
+	install -m 644 lib/lib$(PROGN).so.$(MAJOR).$(MINOR).$(MICRO) $(libdir)/
+	ln -sf lib$(PROGN).so.$(MAJOR).$(MINOR).$(MICRO) $(libdir)/lib$(PROGN).so.$(MAJOR)
+	ln -sf lib$(PROGN).so.$(MAJOR) $(libdir)/lib$(PROGN).so
 
 install-dev: install-lib
-	@mkdir -p $(includedir)/$(PROGN)/
-	@install -m 644 src/classes/*.h $(includedir)/$(PROGN)
+	mkdir -p $(includedir)/$(PROGN)/
+	install -m 644 src/classes/*.h $(includedir)/$(PROGN)
 
 install: setup
-	@test -s /etc/$(PROGN)/profile || \
+	test -s /etc/$(PROGN)/profile || \
 		cp /etc/$(PROGN)/samples/group_keys /etc/$(PROGN)/profile
-	@test -s /etc/$(PROGN)/reboot || \
+	test -s /etc/$(PROGN)/reboot || \
 		cp /etc/$(PROGN)/samples/all_off /etc/$(PROGN)/reboot
-	@udevadm control --reload-rules
-	@$(PROGN) -p /etc/$(PROGN)/profile
-	@test -s /usr/bin/systemd-run && \
+	udevadm control --reload-rules
+	$(PROGN) -p /etc/$(PROGN)/profile
+	test -s /usr/bin/systemd-run && \
 		systemctl daemon-reload && \
 		systemctl enable $(PROGN)-reboot
 
 uninstall-lib:
-	@rm -f $(libdir)/lib$(PROGN).so*
+	rm -f $(libdir)/lib$(PROGN).so*
 
 uninstall-dev:
-	@rm -rf $(includedir)/$(PROGN)
+	rm -rf $(includedir)/$(PROGN)
 
 uninstall:
-	@test -s /usr/bin/systemd-run && \
+	test -s /usr/bin/systemd-run && \
 		systemctl disable $(PROGN)-reboot && \
 		rm $(SYSTEMDDIR)/system/$(PROGN)-reboot.service && \
 		systemctl daemon-reload && \
 		rm -R /etc/$(PROGN)
-	
-	@rm /usr/bin/g213-led
-	@rm /usr/bin/g410-led
-	@rm /usr/bin/g413-led
-	@rm /usr/bin/g512-led
-	@rm /usr/bin/g513-led
-	@rm /usr/bin/g610-led
-	@rm /usr/bin/g910-led
-	@rm /usr/bin/gpro-led
-	@rm /usr/bin/$(PROGN)
-	
-	@rm /etc/udev/rules.d/$(PROGN).rules
-	@udevadm control --reload-rules
+
+	rm /usr/bin/g213-led
+	rm /usr/bin/g410-led
+	rm /usr/bin/g413-led
+	rm /usr/bin/g512-led
+	rm /usr/bin/g513-led
+	rm /usr/bin/g610-led
+	rm /usr/bin/g910-led
+	rm /usr/bin/gpro-led
+	rm /usr/bin/$(PROGN)
+
+	rm /etc/udev/rules.d/$(PROGN).rules
+	udevadm control --reload-rules
